Merging two Linked List using c

saimanoharhm opened the pull request:
<!--
Thank you for sending the PR! We appreciate you spending the time to work on these changes.

Help us understand your motivation by explaining why you decided to make this change.
You can learn more about contributing to NeoAlgo here: https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md

Happy Contributing!

-->

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)
yes
### Description

(Write your answer here.)
I have added code for Merging two Linked List using c
### Checklist

- [ yes] I've read the contribution guidelines.
- [ yes] I've checked the issue list before deciding what to submit.
- [ yes] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

(Write your answer here.)
#2171

saimanoharhm committed:
Adding merging two linked list code

saimanoharhm commented:
@gat786 Could you pls review my PR

gat786 commented:
Add an example of what happens. 
Add a line break at the End Of File(EOF).
@saimanoharhm

saimanoharhm commented:
@gat786 I didn't get you. What Example do you want me to add and break at eof

saimanoharhm commented:
@gat786 Could you pls explain me more eloboratly

vgb-codes commented:
@saimanoharhm Please update the [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md).

HarshCasper commented:
Thanks for making your Pull Request!

PRs would be reviewed after the start of the Coding Period. Thanks!

gat786 commented:
@saimanoharhm An Example will be an example of how one would use your piece of code. Suppose I wrote a program to add two numbers then my example would be 

```
First Number: 12
Second Number: 25
Addition is: 37
```

Like a dummy example of how some one would run your piece of code.

And this [link](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) should clear your mind on why we have a line break at the end of every file.

saimanoharhm commented:
> @saimanoharhm Please update the [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md).

@gat786 I have already updated it. Pls check PR #2198

saimanoharhm commented:
> @saimanoharhm An Example will be an example of how one would use your piece of code. Suppose I wrote a program to add two numbers then my example would be
> 
> ```
> First Number: 12
> Second Number: 25
> Addition is: 37
> ```
> 
> Like a dummy example of how some one would run your piece of code.
> 
> And this [link](https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline) should clear your mind on why we have a line break at the end of every file.

@gat786 Ok I got it! but where do you want me to add it at the end of the program by putting output as a heading or where?

saimanoharhm commented:
> @saimanoharhm Please update the [README.md](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/README.md).

@vgb-codes  I have already updated it and I have made a PR for readme file updation also pls check PR #2198

gat786 commented:
Whereever you feel like it. Poeple usually add it at the bottom of their programs

HarshCasper commented:
@saimanoharhm 

Please make your README updates here itself. Thanks!

saimanoharhm committed:
Updated readme file and code file name

saimanoharhm committed:
Merge branch 'master' of https://github.com/TesseractCoding/NeoAlgo into t

saimanoharhm committed:
Added example output as comment

saimanoharhm commented:
@HarshCasper and @gat786 Could you check it now I have updated readme file and added example output as comment. Pls let me know if further changes needed

saimanoharhm committed:
Add code for Finding a pair of element

saimanoharhm committed:
Merge branch 'master' of https://github.com/TesseractCoding/NeoAlgo into t

saimanoharhm committed:
Update link for Finding a pair of element in readme file

saimanoharhm committed:
Delete both Finding a pair of element code and update of readme file

saimanoharhm commented:
@HarshCasper Could you pls check my PR.

HarshCasper reviewed (changes requested):
Few changes: 

- Resolve the Merge Conflicts. 
- Add Comments above the lines of code wherever possible

saimanoharhm committed:
Merge branch 'master' into t

saimanoharhm committed:
Add comments

saimanoharhm commented:
@HarshCasper Could you pls check it now. I have done all the changes that you asked.

saimanoharhm committed:
Add code for Merging two linked list

saimanoharhm committed:
Update comments

HarshCasper commented on C/ds/Merging_two_Linked_List.c:
```diff
@@ -0,0 +1,127 @@
+This program is about merging two Linkedlist. Here user going to enter values of
+first and second linkedlist. It will merge the
+two linkedlists into one linkedlist by comparing the each and every node values
+and convert into one linkelist in the ascending order.
```
Not gramatically correct: 
- Replace `Linkedlist` with `Linked List`
- Here `the` user `is` going...

HarshCasper reviewed (changes requested):
Use a 4-spaces formatting

saimanoharhm committed:
Update Merging_two_Linked_List.c

HarshCasper reviewed (changes requested):
Few more changes: 
- Don't use variable names like `A`, `B`, `t` and all. Use meaningful variable names
- Share the complexity of the code
- Put spaces after functions 
- Remove some redundant comments

saimanoharhm committed:
Update Merging_two_Linked_List.c

saimanoharhm committed:
Update Merging_two_Linked_List.c
